iec61850serversimulator/src/test/java/com/cgi/iec61850serversimulator/SwitchingMomentCalculatorTest.java

```diff
@@ -0,0 +1,224 @@
+		// 0 = Every day
+		int dayInt = 0;
+		// General time type for both on and off times
+		// 0 = fixed time, 1 = light sensor, 2 = astronomical time
+		// -1 = disabled action
+		int fixedTimeInt = 0;
+		// Optional: Specific time type for on time (-1 = not used)
+		int fixedTimeOn = -1;
+		// Optional: Specific time type for off time (-1 = not used)
+		int fixedTimeOff = -1;
+		LocalTime timeOn = LocalTime.of(12, 00);
+		LocalTime timeOff = LocalTime.of(13, 00);
+		int burningMinutes = 30;
+
+		Schedule schedule = this.getMockSchedule(scheduleNr - 1, relayNr, dayInt, fixedTimeInt, fixedTimeOn,
+				fixedTimeOff, timeOn, timeOff, burningMinutes);
+
+		assertEquals(scheduleNr, schedule.getIndexNumber());
+		assertEquals(relayNr, schedule.getRelayNr());
+		assertEquals(fixedTimeInt, schedule.getTimeOnTypeInt());
+		assertEquals(fixedTimeInt, schedule.getTimeOffTypeInt());
+		assertEquals(timeOn, schedule.getTimeOn());
+		assertEquals(timeOff, schedule.getTimeOff());
+		assertEquals(burningMinutes, schedule.getBurningMinsOn());
+
+	}
+
+	@Test
+	public void calculateEveryDaySwitchingMoments() {
+
+		int scheduleNr = 1;
+		int relayNr = 1;
+		// Trigger day(s), see Enum in GXF LF Energy documentation
+		// 0 = Every day
+		int dayInt = 0;
+		// General time type for both on and off times
+		int fixedTimeInt = 0;
+		// Optional: Specific time type for on time
+		// -1 = using general time type
+		int fixedTimeOn = -1;
+		// Optional: Specific time type for off time
+		// -1 = using general time type
+		int fixedTimeOff = -1;
+		LocalTime timeOn = LocalTime.of(12, 00);
+		LocalTime timeOff = LocalTime.of(13, 00);
+		int burningMinutes = 30;
+		// Optional: Custom schedule
+		Schedule customSchedule = this.getMockSchedule(scheduleNr - 1, relayNr, dayInt, fixedTimeInt, fixedTimeOn,
+				fixedTimeOff, timeOn, timeOff, burningMinutes);
+		boolean useCustomSchedule = false;
+
+		// Initializing mock relay
+		if (!useCustomSchedule) {
+			Schedule schedule = this.getMockLunchTimeSchedule(scheduleNr);
+			Relay relay = this.getMockRelay(relayNr, schedule);
+			Relay[] relays = new Relay[1];
+			relays[0] = relay;
+			this.device.setRelays(relays);
+		} else {
+			Relay relay = this.getMockRelay(relayNr, customSchedule);
+			Relay[] relays = new Relay[1];
+			relays[0] = relay;
+			this.device.setRelays(relays);
+		}
+
+		// Calculating switching moment
+
+		SwitchingMomentCalculator calculator = new SwitchingMomentCalculator();
+		List<SwitchingMoment> switchingMoments = calculator.calculateSwitchingMoments(this.device);
+
+		// Tests
+
+		SwitchingMoment switchingMomentOn = switchingMoments.get(0);
+		SwitchingMoment switchingMomentOff = switchingMoments.get(2);
+		SwitchingMoment switchingMomentNextDayOn = switchingMoments.get(1);
+		SwitchingMoment switchingMomentNextDayOff = switchingMoments.get(3);
+
+		LocalDateTime lunchTime = LocalDateTime.now().withHour(12).withMinute(00).withSecond(0).withNano(0);
```

Comment:
To make it more readable: add "expected" as prefix to the LocalDateTime's you expect.